### PR TITLE
Addition to DeveloperGuide.md

### PR DESCRIPTION
**Pull request by lzf834**
- `Status` Feature
- `FindByStatus` Feature
- `Activity` Feature

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -245,73 +245,60 @@ The following activity diagram summarizes what happens when a user executes a ne
...
-![NAME PLACEHOLDER]()
+**Aspect: Abstracting `Status` attribute**
 
-Step 5. 
+* **Alternative 1 (current choice):** Abstracted class.
+  * Pros: 
+    * Higher Level of abstraction
+    * Changes can be made easily from this class
+  * Cons: 
+    * Adding files to the currently already large amount of files within
```
Comment (joshuayapwj98) [inline docs/DeveloperGuide.md]: Understand this part!

But would the phrasing be better if it was something like:
"Existing layers of abstraction and tangled dependencies make introducing a new attribute for the base Person model difficult."

Nonetheless, the sentence is simple enough!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -245,73 +245,60 @@ The following activity diagram summarizes what happens when a user executes a ne
...
-
-### \[Implemented\] Find By Status feature
+**Aspect: Abstracting into different classes**
 
-#### Implementation
+* **Alternative 1 (current choice):** Abstracted classes.
+    * Pros:
+        * Higher Level of abstraction
+        * Changes can be made easily from this class
+        * Different classes serve different lower-level purposes and are organised into separate packages (e.g. `FindStatusCommandParser` belongs to the `parser` package)
```
Comment (joshuayapwj98) [inline docs/DeveloperGuide.md:292]: This sentence has a good structure. Moreover, it describes the pro and provides an example to support the statement! Perhaps the Abstracting "Status" section can follow this structure!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -329,7 +316,27 @@ The implemented find by class code mechanism is facilitated by `FindByClassCode`
...
+
+**Aspect: Abstracting `activity` attribute**
+
+* **Alternative 1 (current choice):** Abstracted class.
+    * Pros:
+        * Higher Level of abstraction
+        * Changes can be made easily from this class
+        * Ease of accommodating how `activity` attribute can be implemented and added in to the `Person` class as a `Set`
+    * Cons:
+        * Adding files to the currently already large amount of files within
```
Comment (joshuayapwj98) [inline docs/DeveloperGuide.md]: Similar comment to the "Status" portion!

**Review by joshuayapwj98 — DISMISSED** [review]
Well structured and overall well written!

I just added a few comments that could enhance the elaboration of each class! 👍🏼 

I will post this as a "Request Changes" to ensure a second review!!

Otherwise, timely PR!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (lzf834) [reply docs/DeveloperGuide.md]: Alright edited to use suggested sentence instead

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (lzf834) [reply docs/DeveloperGuide.md]: Alright edited to use suggested sentence instead

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -329,7 +316,27 @@ The implemented find by class code mechanism is facilitated by `FindByClassCode`
...
+
+**Aspect: Abstracting `activity` attribute**
+
+* **Alternative 1 (current choice):** Abstracted class.
+    * Pros:
+        * Higher Level of abstraction
+        * Changes can be made easily from this class
+        * Ease of accommodating how `activity` attribute can be implemented and added in to the `Person` class as a `Set`
+    * Cons:
+        * Existing layers of abstraction and tangled dependencies make introducing a new attribute for the base Person model difficult
```
Comment (Fenway17) [inline docs/DeveloperGuide.md:332]: Definitely true. This gets better with time as we get more used to working with the code base.

**Review by Fenway17 — APPROVED** [review]
All looks good!

**Comment by whoisjunhong** [comment]
Merging draft dg for this iteration, changes to be made in future iterations

**Pull request merged**